### PR TITLE
Move zerotrie to utils; change version to 0.1.0

### PR DESCRIPTION
**Pull request by sffc**
To unblock #4024

### CODE REVIEW

**Comment by sffc** [comment]
I should note that other utils that are pre-1.0 are not necessarily up to all ICU4X standards; I consider `zerotrie` at least as polished as `tinystr`. However, I will follow up with an additional commit that adds the boilerplate lints and fixing any issues it finds.

**Comment by robertbastian** [comment]
Please create a stabilisation issue with the checklist.

**Comment by robertbastian** [comment]
I was not taking issue with the code location of zerotrie, more with its overall experimentality. We don't need this PR but I would like to see the checklist.

File: utils/zerotrie/Cargo.toml [hunk 1]
```diff
@@ -5,7 +5,7 @@
 [package]
 name = "zerotrie"
 description = "A data structure that efficiently maps strings to integers"
-version = "0.0.0"
+version = "0.1.0"
```
Comment (robertbastian) [inline utils/zerotrie/Cargo.toml:8]: unpublished crates should be at 0.0.0, otherwise the release will bump to 0.2.0

**Pull request closed without merging**